src/Microsoft.TestPlatform.Client/TestSession/InProcessTestSessionEventsHandler.cs

```diff
@@ -8,17 +8,15 @@
 
 namespace Microsoft.VisualStudio.TestPlatform.Client;
 
-// TODO: Public because there's an attribute definition clash for "MemberNotNullAttribute" and a few other
-// attributes. This should really be internal.
-public class InProcessTestSessionEventsHandler : ITestSessionEventsHandler
+internal class InProcessTestSessionEventsHandler : ITestSessionEventsHandler
 {
     private readonly ITestSessionEventsHandler _testSessionEventsHandler;
 
-    public EventHandler<StartTestSessionCompleteEventArgs?>? StartTestSessionCompleteEventHandler { get; set; }
+    internal EventHandler<StartTestSessionCompleteEventArgs?>? StartTestSessionCompleteEventHandler { get; set; }
```

Comment:
might be easier to keep those members public, they are in an internal class, so they won't be publicly visible, but it will be easier to move the whole class to be public if we ever have the need. It is also how we do it in most of the other code.

Reply:
Yep, I usually do it the same way (i.e. make the class internal and keep members public) but now I was getting some weird build issues and apparently the class being internal was causing some visibility issues with the public members and moving them to internal partially helped. Weird issue, but I managed to fix it in the end and now we do it as you suggested.